src/base/SolutionArray.cpp

```diff
@@ -1139,14 +1276,26 @@ void SolutionArray::append(const vector<double>& state, const AnyMap& extra)
 }
 
 void SolutionArray::save(const string& fname, const string& id, const string& sub,
-                           const string& desc, bool overwrite, int compression)
+                         const string& desc, bool overwrite, int compression,
+                         const string& basis)
 {
     if (m_size < m_dataSize) {
         throw NotImplementedError("SolutionArray::save",
                                   "Unable to save sliced data.");
     }
     size_t dot = fname.find_last_of(".");
     string extension = (dot != npos) ? toLowerCopy(fname.substr(dot + 1)) : "";
+    if (extension == "csv") {
+        if (id != "") {
+            warn_user("SolutionArray::save", "Parameter 'id' not used for CSV output.");
+        }
+        writeEntry(fname, overwrite, basis);
+        return;
+    }
+    if (basis != "") {
+        warn_user("SolutionArray::save",
+            "Species basis '{}' not implemented for HDF5 or YAML output.", basis);
```

Comment:
If I understand correctly, this is here because both mass and molar basis properties are exported? If so, should the message say something more like "basis is not needed for hdf/yaml output"? 

If I don't understand correctly, why isn't this implemented? 🤪

Reply:
YAML and HDF5 use the 'native' basis by default; the non-native export/import options aren't implemented (I don't see a need for that, especially as they cannot be optimized). I added the flag to support the basis selection for CSV, where this switch is a lot more useful.